doc/neps/nep-0051-scalar-representation.rst

```diff
@@ -0,0 +1,332 @@
+
+    np.True_
+    np.float64(3.0)
+    np.int64(34)
+
+and so on.  This will also mean that documentation and output in
+Jupyter notebook cells will often show the type information intact.
+
+``np.longdouble`` and ``np.clongdouble`` will print with single quotes::
+
+    np.longdouble('3.0')
+
+to allow round-tripping.  Addtionally to this change, ``float128`` will
+now always be printed as ``longdouble`` since the old name gives a wrong
+impression of precision.
+
+Backward compatibility
+======================
+
+We expect that most workflows will not be affected as only printing
+changes.  In general we believe that informing users about the type
+they are working with outweighs the need for adapting printing in
+some instances.
+
+The NumPy test suite includes code such as ``decimal.Decimal(repr(scalar))``.
+This code needs to be modified to use the ``str()``.
+
+An exception to this are downstream libraries with documentation and
+especially documentation testing.
+Since the representation of many values will change, in many cases
+the documentation will have to be updated.
+This is expected to require larger documentation fixups in the mid-term.
+
+Further, it may be necessary to adept tools for doctest testing to
+allow approximate value checking for the new representation.
+
+Changes to ``arr.tofile()``
+---------------------------
+``arr.tofile()`` currently stores values as ``repr(arr.item())`` when in text
+mode.  This is not always ideal also since that may include a conversion to
+Python.
+One issue is that this would start saving longdouble as
+``np.longdouble('3.1')`` which is clearly not desired.  We expect that this
+method is rarely used for object arrays.  For string arrays, using the ``repr``
+also leads to storing ``"string"`` or ``b"string"`` which seems rarely desired.
+
+The proposal is to change the default (back) to use ``str`` rather than
+``repr``.  If ``repr`` is desired, users will have to pass ``fmt=%r``.
+
+
+Detailed description
+====================
+
+This NEP proposes to change the represenatation for NumPy scalars to:
+
+* ``np.True_`` and ``np.False_`` for booleans
+* ``np.scalar(<value>)``, i.e. ``np.float64(3.0)`` for all numerical dtypes.
+* The value for ``np.longdouble`` and ``np.clongdouble`` will be given in quotes:
+  ``np.longdouble('3.0')``.  This ensures that it can always roundtrip correctly
+  and matches the way that ``decimal.Decimal`` behaves.
+  For these two the size based name such as ``float128`` will not be used
+  as it is platform dependend and misleading.
+* ``np.str_("string")`` and ``np.bytes_(b"byte_string")`` for string dtypes.
+* ``np.void((3, 5), dtype=[('a', '<i8'), ('b', 'u1')])`` (similar to arrays),
+  this will be valid syntax to recreate the scalar.
+  Unlike arrays, the representation should round-trip correctly, so longdouble
+  values will be quoted and other values never be truncated.
+
+Where booleans are printed as their singletons since this is more concise.
```

Comment:
I'm not sure what we're trying to say here. Is this to highlight the (proposed) difference between `str(<numpy.boolean scalar>)` and `repr(<numpy.boolean scalar>`?

Reply:
It was to point out that it isn't `np.bool_(True)`.  Moved it as a bracket above to say that these are the singletons and removed the following sentence as well.  (This was a bit of a leftover for when I wanted to stress the general pattern of `np.type(value)`).